### PR TITLE
mmedit -> mmagic

### PR DESCRIPTION
**Pull request by huayuan4396**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily receiving feedbacks. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Please describe the motivation of this PR and the goal you want to achieve through this PR.

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit tests to ensure the correctness.
3. If the modification has a dependency on downstream projects of a newer version, this PR should be tested with all supported versions of downstream projects.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

**Review by RunningLeon — COMMENTED** [review]
could fix lint

**Comment by huayuan4396** [comment]
Now the lint check is failed due to the modified mmagic path, it can be ignored.

File: mmdeploy/codebase/mmagic/deploy/super_resolution_model.py [hunk 1]
```diff
@@ -194,25 +197,27 @@ def forward(self,
...
+
         if hasattr(self.data_preprocessor, 'destructor'):
             inputs = self.data_preprocessor.destructor(
                 inputs.to(self.data_preprocessor.input_std.device))
+
         outputs = []
         for i in range(inputs.shape[0]):
             output = self.wrapper.invoke(inputs[i].permute(
-                1, 2, 0).contiguous().detach().cpu().numpy())
+                1, 2, 0).contiguous().detach().cpu().numpy() * 255.)
```
Comment (RunningLeon) [inline mmdeploy/codebase/mmagic/deploy/super_resolution_model.py:208]: why do we need to multiply by 255? Is there any corresponding code in mmagic?

Comment (huayuan4396) [reply mmdeploy/codebase/mmagic/deploy/super_resolution_model.py:208]: The input should be in range 0-255, or the model prediction is wrong.

Comment (RunningLeon) [reply mmdeploy/codebase/mmagic/deploy/super_resolution_model.py:208]: but we have normalization in pipeline.json, this should do the work. Could double check here?

` ``
                   {
                        "type": "Normalize",
                        "mean": [
                            0.0,
                            0.0,
                            0.0
                        ],
                        "std": [
                            255.0,
                            255.0,
                            255.0
                        ],
                        "to_rgb": false
                    },
```

**Comment by RunningLeon** [comment]
@AllentDan Hi, any comments on this PR?

**Comment by huayuan4396** [comment]
> Yeah. I am also waiting for the reply for `multiply by 255`. And the following commands throw me an error:
> 
> ```shell
> python tools/deploy.py configs/mmagic/super-resolution/super-resolution_tensorrt_dynamic-32x32-512x512.py ../mmagic/configs/srcnn/srcnn_x4k915_1xb16-1000k_div2k.py https://download.openmmlab.com/mmediting/restorers/srcnn/srcnn_x4k915_1x16_1000k_div2k_20200608-4186f232.pth data/Set5/LRbicx4/baby.png --work-dir work_dirs/tensorrt/mmagic/srcnn --device cuda --show --dump-info
> ```
> 
> error:
> 
> ```
> Expected all tensors to be on the same device, but found at least two devices, cuda:0 and cpu!
> ```

I think this may be a bug of mmagic, I add the device setting in mmagic and it works fine. I have report this issue in mmagic, no reply yet.

File: mmdeploy/codebase/mmagic/deploy/super_resolution_model.py [hunk 2]
```diff
@@ -121,30 +121,33 @@ def forward(self,
             get_root_logger().warning(f'expect input device {self.device}'
                                       f' but get {lq.device}.')
         lq = lq.to(self.device)
+
         batch_outputs = self.wrapper({self.input_name:
-                                      lq})[self.output_names[0]].to('cpu')
+                                      lq})[self.output_names[0]].to(self.device)
```
Comment (AllentDan) [inline mmdeploy/codebase/mmagic/deploy/super_resolution_model.py:126]: Generally, we do `destruct` on CPU devices. So, there is no need to change `cpu` to `self.device`.

Comment (huayuan4396) [reply mmdeploy/codebase/mmagic/deploy/super_resolution_model.py:126]: > Generally, we do `destruct` on CPU devices. So, there is no need to change `cpu` to `self.device`.

OK, I will change that back.

**Review by RunningLeon — APPROVED** [review]
LGTM

**Comment by AllentDan** [comment]
Running `tools/test.py configs/mmagic/super-resolution/super-resolution_sdk_dynamic.py ../mmagic/configs/srcnn/srcnn_x4k915_1xb16-1000k_div2k.py --model work_dirs/tensorrt/mmagic/srcnn --device cuda` throw me an error `RuntimeError: Expected all tensors to be on the same device, but found at least two devices, cuda:0 and cpu`

**Comment by huayuan4396** [comment]
Modify `mmagic/models/data_preprocessors/data_preprocessor.py` line 725 from:

```python
mean = self.mean.view(target_shape)
std = self.std.view(target_shape)
batch_tensor = batch_tensor * std + mean
```

to:

```python
mean = self.mean.view(target_shape).to(batch_tensor.device)
std = self.std.view(target_shape).to(batch_tensor.device)
batch_tensor = batch_tensor * std + mean
```

and it works fine.

File: mmdeploy/codebase/mmagic/deploy/super_resolution_model.py [hunk 3]
```diff
@@ -65,8 +65,8 @@ def _init_wrapper(self, backend: Backend, backend_files: Sequence[str],
             **kwargs)
 
     def convert_to_datasample(
-            self, predictions: EditDataSample, data_samples: EditDataSample,
-            inputs: Optional[torch.Tensor]) -> List[EditDataSample]:
+            self, predictions: DataSample, data_samples: DataSample,
```
Comment (AllentDan) [inline mmdeploy/codebase/mmagic/deploy/super_resolution_model.py:68]: `predictions` is already a DataSample. So the name `convert_to_datasample` is not appropriate. Maybe we can pass `pred_img` as an argument or we can rename the function.

Comment (huayuan4396) [reply mmdeploy/codebase/mmagic/deploy/super_resolution_model.py:68]: MMagic iterate the `forward` output like:
```python
for data_sample in data_samples:
    pred_img = data_sample.pred_img
    ...
```
So how about we rename this function from `convert_to_datasample` to `wrap_in_datasample`?

Comment (AllentDan) [reply mmdeploy/codebase/mmagic/deploy/super_resolution_model.py:68]: @RunningLeon any suggestion? Anyway, the docstring of the function should be updated.

**Comment by huayuan4396** [comment]
The `convert_to_datasample` should output a list of `DataSample`, have renamed it to `convert_to_datasample_list` and remove this function in `SDKEnd2EndModel`, since it can be derived from `End2EndModel`.

**Comment by AllentDan** [comment]
Please update the docstring of `convert_to_datasample_list`

**Review by AllentDan — APPROVED** [review]
LGTM

**Pull request merged**